core/local/local.go

```diff
@@ -70,7 +70,11 @@ func NewExecutionScheduler(runner lib.Runner, logger *logrus.Logger) (*Execution
 	executors := make([]lib.Executor, 0, len(executorConfigs))
 	// Only take executors which have work.
 	for _, sc := range executorConfigs {
-		if !sc.HasWork(executionState.Options.ExecutionSegment) {
+		if !sc.HasWork(options.ExecutionSegment) {
+			logger.Warnf(
+				"Executor '%s' is disabled for segment %s due to lack of work!",
+				sc.GetName(), options.ExecutionSegment,
+			)
```

Comment:
I am wondering if instead of 
```
WARN[0000] Executor 'shared_iters1' is disabled for segment 1/2:3/4 due to lack of work!
   executor: local
     output: -
     script: 1295.js

  execution: (25.00%) 3 executors, 2 max VUs, 10m30s max duration (incl. graceful stop):
           * constant_arr_rate: 0.75 iterations/s for 20s (maxVUs: 1, gracefulStop: 30s)
           * shared_iters1: 0 iterations shared among 0 VUs (maxDuration: 10m0s, gracefulStop: 30s)
           * shared_iters2: 1 iterations shared among 1 VUs (maxDuration: 10m0s, gracefulStop: 30s)


running (00m20.1s), 0/2 VUs, 15 complete and 0 interrupted iterations
constant_arr_rate [======================================] 0.75 iters/s, 1 out of 1 VUs active done!
shared_iters2 [======================================] 1/1 shared iters among 1 VUs done!


    data_received........: 0 B 0 B/s
    data_sent............: 0 B 0 B/s
    iteration_duration...: avg=33.41µs min=4.91µs med=24.81µs max=87.14µs p(90)=74.71µs p(95)=82.18µs
    iterations...........: 15  0.745541/s
    vus..................: 1   min=1 max=1

```
it would be better to be just before the progressbars 
```
   executor: local
     output: -
     script: 1295.js

  execution: (25.00%) 3 executors, 2 max VUs, 10m30s max duration (incl. graceful stop):
           * constant_arr_rate: 0.75 iterations/s for 20s (maxVUs: 1, gracefulStop: 30s)
           * shared_iters1: 0 iterations shared among 0 VUs (maxDuration: 10m0s, gracefulStop: 30s)
           * shared_iters2: 1 iterations shared among 1 VUs (maxDuration: 10m0s, gracefulStop: 30s)

WARN[0000] Executor 'shared_iters1' is disabled for segment 1/2:3/4 due to lack of work!

running (00m20.1s), 0/2 VUs, 15 complete and 0 interrupted iterations
constant_arr_rate [======================================] 0.75 iters/s, 1 out of 1 VUs active done!
shared_iters2 [======================================] 1/1 shared iters among 1 VUs done!


    data_received........: 0 B 0 B/s
    data_sent............: 0 B 0 B/s
    iteration_duration...: avg=33.41µs min=4.91µs med=24.81µs max=87.14µs p(90)=74.71µs p(95)=82.18µs
    iterations...........: 15  0.745541/s
    vus..................: 1   min=1 max=1

```

Reply:
it will be a bit better, yeah.... though that would require some very refactoring that I'm not sure I want to do right now... basically, we'd have to reorder a lot of `cmd/run.go` to achieve this :disappointed:

Reply:
okay let's leave it for a future PR